nav2_costmap_2d/src/footprint_subscriber.cpp

```diff
@@ -23,53 +23,78 @@ namespace nav2_costmap_2d
 
 FootprintSubscriber::FootprintSubscriber(
   nav2_util::LifecycleNode::SharedPtr node,
-  const std::string & topic_name)
+  const std::string & topic_name,
+  const double & footprint_timeout)
 : FootprintSubscriber(node->get_node_base_interface(),
     node->get_node_topics_interface(),
     node->get_node_logging_interface(),
-    topic_name)
+    node->get_node_clock_interface(),
+    topic_name, footprint_timeout)
 {}
 
 FootprintSubscriber::FootprintSubscriber(
   rclcpp::Node::SharedPtr node,
-  const std::string & topic_name)
+  const std::string & topic_name,
+  const double & footprint_timeout)
 : FootprintSubscriber(node->get_node_base_interface(),
     node->get_node_topics_interface(),
     node->get_node_logging_interface(),
-    topic_name)
+    node->get_node_clock_interface(),
+    topic_name, footprint_timeout)
 {}
 
 FootprintSubscriber::FootprintSubscriber(
   const rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
   const rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
   const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
-  const std::string & topic_name)
+  const rclcpp::node_interfaces::NodeClockInterface::SharedPtr node_clock,
+  const std::string & topic_name,
+  const double & footprint_timeout)
 : node_base_(node_base),
   node_topics_(node_topics),
   node_logging_(node_logging),
-  topic_name_(topic_name)
+  node_clock_(node_clock),
+  topic_name_(topic_name),
+  footprint_received_time_(0.0),
+  footprint_timeout_(footprint_timeout)
 {
   footprint_sub_ = rclcpp::create_subscription<geometry_msgs::msg::PolygonStamped>(node_topics_,
       topic_name, rclcpp::SystemDefaultsQoS(),
       std::bind(&FootprintSubscriber::footprint_callback, this, std::placeholders::_1));
 }
 
 bool
-FootprintSubscriber::getFootprint(std::vector<geometry_msgs::msg::Point> & footprint)
+FootprintSubscriber::getFootprint(
+  std::vector<geometry_msgs::msg::Point> & footprint,
+  rclcpp::Time & stamp)
 {
   if (!footprint_received_) {
     return false;
   }
 
   footprint = toPointVector(
     std::make_shared<geometry_msgs::msg::Polygon>(footprint_->polygon));
+  stamp = footprint_->header.stamp;
+
+  if (stamp - footprint_received_time_ > rclcpp::Duration(1.0, 0.0)) {
```

Comment:
Seems like the duration value should be a parameter. Is this supposed to be the `footprint_timeout_` parameter?   On that note, the `footprint_timeout_` is not being used anywhere in this code. Also, aren't the `stamp` and `footprint_received_time_` the same thing here, looks like they are both assigned form the latest footpint msg. I want to verify the intended logic here.

It would seem to me you want that if the latest footprint time stamp is older than a specified timeout then you return false to the user. I'm not sure we need to do that in this code, but just give them the stamp and let the user decide what to do with it.

Reply:
New commit pushed, sorry about that, I'm a compulsive `push`er, should make sense now